src/SprykerSdk/Acp/Console/RegisterConsole.php

```diff
@@ -134,25 +136,33 @@ protected function register(InputInterface $input, OutputInterface $output): int
         $registerRequestTransfer->setManifestPath($input->getOption(AppManifestValidateConsole::MANIFEST_PATH));
         $registerRequestTransfer->setConfigurationFile($input->getOption(AppConfigurationValidateConsole::CONFIGURATION_FILE));
         $registerRequestTransfer->setTranslationFile($input->getOption(AppTranslationValidateConsole::TRANSLATION_FILE));
+        $registerRequestTransfer->setAcpApiFile($input->getOption('acpApiFile'));
 
-        $registerResponseTransfer = $this->getFacade()->registerApp($registerRequestTransfer);
+        if ($input->getOption('dry-run')) {
+            $output->writeln('Running in dry-run mode. No actual changes will be made.');
 
-        $message = 'App successfully registered or updated in ACP.';
-        $responseCode = static::CODE_SUCCESS;
+            $requestBody = $this->getFacade()->getRegistrationRequestBody($registerRequestTransfer);
```

Comment:
I would not add a facade method for this and would rather pass the dry-run option into the model.